### ISSUE TITLE
Ability to remove a node

### ISSUE THREAD

**Issue by viq** [juanfont/headscale#52]
Currently nodes can only be added. I'd like to be able to remove nodes that don't exist anymore (without going into SQL myself).

**Comment by juanfont** [comment]
Done! Although you might have to reload headscale for changes to take effect. We will be working on that next.

**Comment by viq** [comment]
Seems also the same is true for ephemeral nodes - they get cleaned up only on headscale restart.

**Comment by juanfont** [comment]
That really sounds like something is being cached. Are you serving this
through Cloudflare by any chance?

**Comment by viq** [comment]
No, purely my infra: couple OpenBSD boxes on libvirt on a dedicated server,
and all the pieces are running on them. The only possible caching is via
nginx, and I thought I disabled it in config.

**Comment by juanfont** [comment]
Can you make sure all the requests arriving to nginx are fwded to
Headscale? It sounds very weird...

**Comment by viq** [comment]
I just verified with `tailscaled` on a couple machines connecting directly to `headscale`'s port `8000`, and even when `headscale` logged removal of the ephemeral node, all other machines still show it listed in `tailscale status`.

**Comment by viq** [comment]
Though I'll add this is with `0.3.4`, I didn't try building `master`. Would
you expect different result there?

**Comment by juanfont** [comment]
I will tag a new release as soon as https://github.com/juanfont/headscale/pull/63 is merged. Apologies for the late response, busy week :(

**Comment by viq** [comment]
Thanks :)
No need to apologise, those are very quick responses for my standards ;)

**Comment by juanfont** [comment]
This is fixed now :)

**Comment by djkazic** [comment]
I'm seeing this behavior on headscale `0.22.3`.

I'd register a node A, then once I see it in `headscale nodes list`, remove it with `headscale nodes delete -i 5`. Confirmed it's gone from the list, but `tailscale status` on my other machines would still have it there, and it could continue to network.

Only when I restart the tailscale daemon does the connection drop. If I restart the tailscale app of the removed node, similarly, it updates then.

I looked at the code and there doesn't seem to be any polling now, but rather usage of a mapper. Does that mapper suffer from the same issue as what was seen here?

Context: I wanted to write some code that would be able to eject a node from the tailnet, but this doesn't seem possible.
